itest/send_test.go

```diff
@@ -13,6 +16,50 @@ import (
 // testBasicSend tests that we can properly send assets back and forth between
 // nodes.
 func testBasicSend(t *harnessTest) {
+	var (
+		ctxb = context.Background()
+		wg   = sync.WaitGroup{}
+	)
+
+	// Subscribe to receive assent send events from primary taro node.
+	eventNtfns, err := t.tarod.SubscribeSendAssetEventNtfns(
+		ctxb, &tarorpc.SubscribeSendAssetEventNtfnsRequest{},
+	)
+	require.NoError(t.t, err)
+
+	// Test to ensure that we execute the transaction broadcast state.
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		foundSendSates := make(map[string]struct{})
+		for {
+			event, err := eventNtfns.Recv()
+
+			// Break if we get an EOF, which means the stream was
+			// closed.
+			if err != nil &&
+				strings.Contains(err.Error(), io.EOF.Error()) {
+
+				break
+			}
+
+			// If err is not EOF, then we expect it to be nil.
+			require.NoError(t.t, err)
+
+			// Check for the transaction broadcast state.
+			switch eventTyped := event.Event.(type) {
+			case *tarorpc.SendAssetEvent_ExecuteSendStateEvent:
+				executeSendStateEvent := eventTyped.ExecuteSendStateEvent
+				foundSendSates[executeSendStateEvent.SendState] = struct{}{}
+			}
+		}
+
+		// Ensure that we have found the broadcast state.
+		_, ok := foundSendSates["SendStateBroadcast"]
+		require.True(t.t, ok)
```

Comment:
IIUC failing the test in another goroutine at time doesn't actually work?

Reply:
Adding `require.True(t.t, false)` to the goroutine fails the test with:

```
    send_test.go:62: 
        	Error Trace:	/home/user/dev/taro/itest/send_test.go:62
        	            				/home/user/dev/taro/itest/asm_amd64.s:1594
        	Error:      	Should be true
        	Test:       	TestTaroDaemon/basic_send
    --- FAIL: TestTaroDaemon/basic_send (4.24s)
```

but is there some doubt as to whether this test will fail reliably?

Reply:
I think it's okay to fail within a goroutine as long as you make sure it actually runs. So I think we need to add a `wg.Wait()` somewhere.

Reply:
`wg.Wait()` added :+1: